From 5b3b4446b20dee60f1be110378e98db05b94f0d3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jakub=20Van=C4=9Bk?= <linuxtardis@gmail.com>
Date: Tue, 28 Dec 2021 15:43:10 +0100
Subject: [PATCH 656/828] dtoverlays: Enable cam1_clock when using tc358743 or
 irs1125

This fixes a regression introduced in 131f1322039284932ccb601a5cffdd9ca5d36d96
(see also https://github.com/raspberrypi/linux/issues/4791).
The tc358743 driver refused to bind to the device. The irs1125
driver is likely behaving similarly.

The new unified cam1_clk node that represents the fixed on-board
oscillator is marked as disabled by default. These overlays didn't
expect this and so the clock nodes were stuck in disabled state.

This commit just adds the required status = "okay" line. Other sensor
drivers do this too.
---
 arch/arm/boot/dts/overlays/irs1125-overlay.dts  | 1 +
 arch/arm/boot/dts/overlays/tc358743-overlay.dts | 1 +
 2 files changed, 2 insertions(+)

--- a/arch/arm/boot/dts/overlays/irs1125-overlay.dts
+++ b/arch/arm/boot/dts/overlays/irs1125-overlay.dts
@@ -75,6 +75,7 @@
 	clk_frag: fragment@5 {
 		target = <&cam1_clk>;
 		__overlay__ {
+			status = "okay";
 			clock-frequency = <26000000>;
 		};
 	};
--- a/arch/arm/boot/dts/overlays/tc358743-overlay.dts
+++ b/arch/arm/boot/dts/overlays/tc358743-overlay.dts
@@ -78,6 +78,7 @@
 	clk_frag: fragment@6 {
 		target = <&cam1_clk>;
 		__overlay__ {
+			status = "okay";
 			clock-frequency = <27000000>;
 		};
 	};
